build.gradle

```diff
@@ -90,14 +90,12 @@ subprojects { subproject ->
             api "org.apache.groovy:groovy:$groovyVersion"
             api "org.slf4j:slf4j-api:$slf4jVersion"
 
-            testImplementation "org.spockframework:spock-core:${spockVersion}", {
+            testImplementation "org.spockframework:spock-core:$spockVersion", {
```

Comment:
Looks like Spock doesn't use groovy-all anymore.  Can you get rid of the exclude? 

https://repo1.maven.org/maven2/org/spockframework/spock-core/2.3-groovy-4.0/spock-core-2.3-groovy-4.0.pom

Reply:
@codeconsole Yes, there are many things to improve. Once you start it's hard to limit yourself. I had actually removed that exclusion and done some other stuff, but then I thought of @jamesfredley comment about atomic git commits 😄 . I mean, I could go on for a day in this project. I'll add some more commits to this PR.

Reply:
thanks for getting rid of it. looks even better now 😄